Add script to update german shool holidays

KristjanESPERANTO opened the pull request:
I have problems with hc to update the shool holidays, so I wrote another script.

Side note: In the medium term, I would like to get PH and SH from the API of https://www.openholidaysapi.org, but I would only make the effort if PRs are accepted here again. I don't want to do the work for nothing.

KristjanESPERANTO commented on scripts/update_german_sh.mjs:
```diff
@@ -0,0 +1,176 @@
+      const endMonth = endDate.getMonth() + 1;
+      const endDay = endDate.getDate();
+
+      try {
```
That is not necessary. I have just tested it with the year 2026.

HolgerJeromin replied:
This depends on the api data? The oneliner
`holidayData[entry.year] ??= []`
is not too ugly

KristjanESPERANTO replied:
Line 98 sets the year array to `[]`. So it is not necessary to intercept this here. Maybe we could refactor it, but as the script is now, it shouldn't be a problem.

HolgerJeromin replied:
oh, yes. 
Reviewing code on mobile has its downside. :-)

KristjanESPERANTO replied:
All good. Thanks for the review! Better to ask more than too little :slightly_smiling_face:

ypid reviewed (dismissed):
I have only peeked at the script yet but I still want to leave a few comments because the licensing stuff is interesting for full #485 but not required.

KristjanESPERANTO commented on scripts/update_german_sh.mjs:
```diff
@@ -0,0 +1,173 @@
+    // Remove the last newline character to satisfy the YAML linter
+    yamlString = yamlString.slice(0, -1);
+    // Replace all double quotes with single quotes
+    yamlString = yamlString.replace(/"/g, "'");
```
Good point. But when I use prettier, yamllint complains about a lot of other formatting things. I have now simplified this part a bit without regex.

KristjanESPERANTO commented on scripts/update_german_sh.mjs:
```diff
@@ -0,0 +1,173 @@
+/*
+  This script fetches the German school holidays from the ferien-api [1][2] and
```
Done :slightly_smiling_face:

KristjanESPERANTO commented on scripts/update_german_sh.mjs:
```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2025 Kristjan ESPERANTO
```
No, I just used other files as template :slightly_smiling_face:

ypid replied:
Good point. You are right, "Copyright (c)" was still in the repo. I have updated #487 to remove all left over occurrences.

KristjanESPERANTO commented:
I'm now rebasing and fixing the commit message.

KristjanESPERANTO commented:
I just rebased the PR and changed the base branch from master to develop.

@ypid I would like to merge this into develop, but I need your review approval :slightly_smiling_face:

KristjanESPERANTO commented:
Just removed an unused import in the new script and completed the data.
